renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -250,9 +398,48 @@ rdcarray<EventUsage> QueryModifyingEvents(WrappedOpenGL *driver, GLPixelHistoryR
   return modEvents;
 }
 
+void QueryPostModPixelValues(WrappedOpenGL *driver, GLPixelHistoryResources &resources,
+                             const rdcarray<EventUsage> &modEvents, int x, int y,
+                             rdcarray<PixelModification> &history)
+{
+  driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+  driver->glClear(eGL_COLOR_BUFFER_BIT | eGL_DEPTH_BUFFER_BIT | eGL_STENCIL_BUFFER_BIT);
+  driver->ReplayLog(0, modEvents[0].eventId, eReplay_WithoutDraw);
+
+  for(size_t i = 0; i < modEvents.size(); i++)
+  {
+    GLint savedReadFramebuffer, savedDrawFramebuffer;
+    driver->glGetIntegerv(eGL_DRAW_FRAMEBUFFER_BINDING, &savedDrawFramebuffer);
+    driver->glGetIntegerv(eGL_READ_FRAMEBUFFER_BINDING, &savedReadFramebuffer);
+    // bind our own framebuffer to save the pixel values
+    driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+    driver->ReplayLog(modEvents[i].eventId, modEvents[i].eventId, eReplay_Full);
+
+    // read the post mod pixel value into the history event
+    ModificationValue modValue;
+    PixelValue pixelValue;
+    driver->glReadPixels(x, y, 1, 1, eGL_RGBA, eGL_FLOAT, (void *)pixelValue.floatValue.data());
+    driver->glReadPixels(x, y, 1, 1, eGL_DEPTH_COMPONENT, eGL_FLOAT, (void *)&modValue.depth);
+    driver->glReadPixels(x, y, 1, 1, eGL_STENCIL_INDEX, eGL_INT, (void *)&modValue.stencil);
+    modValue.col = pixelValue;
+
+    history[i].postMod = modValue;
+
+    // restore the capture's framebuffer
+    driver->glBindFramebuffer(eGL_DRAW_FRAMEBUFFER, savedDrawFramebuffer);
+    driver->glBindFramebuffer(eGL_READ_FRAMEBUFFER, savedReadFramebuffer);
```

Comment:
There's an issue here - when replaying the capture in this pass we're not actually writing any of the draws to the original framebuffers. This means that subsequent passes depending on the results of earlier passes may be completely wrong because the data isn't there - worst case it could lead to a GPU crash if the data affects control flow.

It also has a minor problem here that we're fetching the postmod data at 'full' precision because our offscreen framebuffer is full floats. Part of the value of the postmod data is that it's at 'real' precision - e.g. if the texture is RGBA8 then this shows any quantisation that happens from the shader output value.

I think a better solution here is for each draw to replay it as normal, but then use a single pixel framebuffer blit to copy the output (after quantisation) to our framebuffer. I would also recommend that you blit each pixel to a new slot in the framebuffer and then do all the reads at the end - as this stands at the moment it's syncing the GPU every draw to readback the results, but you can do all of this work pipelined.